events/management/commands/export_organizations.py

```diff
@@ -0,0 +1,68 @@
+from django.core.management.base import BaseCommand
+from django_orghierarchy.models import Organization
+
+
+class Command(BaseCommand):
+    help = "Export organization data"
+
+    def add_arguments(self, parser):
+        parser.add_argument("--filename", help="Export filename")
+        parser.add_argument("--delimiter", default=";", help="CSV delimiter")
+
+    def _iterate_tree(self, objects, delimiter, depth=0, max_depth=0):
+        tree = []
+
+        for obj in objects:
+            children = obj.children.all()
+
+            line = depth * delimiter + f"{obj.name} ({obj.id})"
+            tree.append(line)
+
+            if children:
+                subtree, max_depth = self._iterate_tree(
+                    children, delimiter, depth=depth + 1, max_depth=max_depth
+                )
+                tree.extend(subtree)
+
+        if max_depth < depth:
+            max_depth = depth
+
+        return tree, max_depth
+
+    def make_valid_csv(self, rows, max_depth, delimiter):
+        """
+        Make sure all lines contain newline characters and the same amount
+        of delimiters. Add header.
+        """
+
+        header = ""
+        for i in range(max_depth + 1):
+            header += f"Level {i+1}{delimiter}"
+
+        fixed_rows = [header.rstrip(delimiter) + "\n"]
+
+        for row in rows:
+            depth_after_node = max_depth - row.count(delimiter)
+
+            new_row = row + depth_after_node * delimiter + "\n"
+            fixed_rows.append(new_row)
+
+        return fixed_rows
+
+    def _write_csv(self, filename, delimiter):
+
+        orgs = Organization.objects.filter(parent__isnull=True)
+        rows, max_depth = self._iterate_tree(orgs, delimiter)
+
+        rows = self.make_valid_csv(rows, max_depth, delimiter)
+
+        with open(filename, "w") as csvfile:
+            csvfile.writelines(rows)
```

Comment:
Just FYI, [Python also has](https://docs.python.org/3/library/csv.html#csv.writer) `csv.writer` which some dialect related issues. 

```
writer = csv.writer(csvfile, delimiter=delimiter)
writer.writerows(list_of_rows)  # A single row in this list would be a list of strings
```

Reply:
Am aware of `csv` module. However, formatting the lines with correct amount of columns was the hardest part in this so I felt like it's easier to work with lines as strings directly.